### ISSUE TITLE
azuread_application 'type' flag results in broken configuration loop

### ISSUE THREAD

**Issue by PirateBread** [hashicorp/terraform-provider-azuread#236]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version
+ Terraform v0.12.24
+ provider.azuread v0.8.0
+ provider.azurerm v1.42.0
+ provider.random v2.2.1

### Affected Resource(s)

* `azuread_application`

### Terraform Configuration Files

```
resource "azuread_application" "client-app" {
  name = "client-app"
  reply_urls = ["https://client-app",]
  public_client = "true"
  oauth2_allow_implicit_flow = "true"
}

```
#
### Expected Behavior

Terraform should ignore the type field.

> App registrations aren't limited to being either a web app/API or a native app. You can use the same app registration for all of these apps by registering the respective redirect URIs.

https://docs.microsoft.com/en-us/azure/active-directory/develop/app-registrations-training-guide-for-app-registrations-legacy-users

### Actual Behavior

1. Application is created correctly on first Terraform Apply
2. Second Terraform Apply causes following changes:

```
azuread_application.client-app will be updated in-place
  ~ resource "azuread_application" "client-app" {
        homepage                   = "https://client-app"
        name                       = "client-app"
        oauth2_allow_implicit_flow = true
        public_client              = true
        reply_urls                 = [
            "https://client-app",
        ]
      **~ type                       = "native" -> "webapp/api"** ### why?
    }
```

3. Third Terraform Apply causes the following changes:
```
azuread_application.client-app will be updated in-place
  ~ resource "azuread_application" "client-app" {
        homepage                   = "https://client-app"
        name                       = "client-app"
        oauth2_allow_implicit_flow = true
      ~ public_client              = false -> true ### It lost this flag from the previous apply.
        reply_urls                 = [
            "https://client-app",
        ]
        type                       = "webapp/api"
    }
```
When the 2nd Terraform Apply runs and sets the application to "webapp/api" - It causes the Application to drop the "public_client" flag. (confirmed in Portal)  This causes Terraform to try and set the flag back.

4. At this point, running further TF Applys don't show any immediate changes, however within 2-10 minutes, the application will be reverted back to "Native" web app at which point you can go back to Step 2 and the loop begins again.

### Steps to Reproduce

1. Create App Registration with reply URLS and public_client=true
2. Run Terraform Apply multiple times

### Important Factoids
This is how I have stopped this loop:
`    ignore_changes = [type, ]`

**Comment by manicminer** [comment]
Hi @PirateBread, thanks for raising this. I've looked into the provider logic and I don't believe we're effecting this behavior. This looks to be a side effect of the API we're using (AAD Graph) being unable to support new-style reply URLs / redirect URIs and if you specify any, it behaves in the way you're experiencing where the (deprecated) publicClient property is reset. Terraform responds accordingly by trying in vain to enforce the configuration you specify. We're deriving the `type` attribute from the value of `public_client` - which is perhaps not super transparent but is the reason for the flipflopping of that.

This will be a much better experience once we have implemented app registrations via the newer (MS Graph) API.
